plugins/hw_event/hw_event_plugin.go

```diff
@@ -0,0 +1,207 @@
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/redhat-cne/cloud-event-proxy/pkg/common"
+	"github.com/redhat-cne/cloud-event-proxy/pkg/restclient"
+	cneevent "github.com/redhat-cne/sdk-go/pkg/event"
+	"github.com/redhat-cne/sdk-go/pkg/pubsub"
+	"github.com/redhat-cne/sdk-go/pkg/types"
+	v1amqp "github.com/redhat-cne/sdk-go/v1/amqp"
+	v1event "github.com/redhat-cne/sdk-go/v1/event"
+	v1pubsub "github.com/redhat-cne/sdk-go/v1/pubsub"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	resourceAddress string        = "/hw-event/cpu"
+	eventInterval   time.Duration = time.Second * 5
+	apiPath         string        = "/api/cloudNotifications/v1/"
+	apiAddr         string        = "localhost:8080"
+	localAPIAddr    string        = "localhost:9088"
+	localAPIPath    string        = "ack/event"
+	hwSubUrl        string        = "https://10.19.109.249/redfish/v1/EventService/Subscriptions/"
+	hwEventTypes    []string      = []string{"StatusChange", "ResourceUpdated", "ResourceAdded", "ResourceRemoved", "Alert"}
+)
+
+type HttpHeader struct {
+	ContentType     string `json:"Content-Type" example:"application/json"`
+	ContentLanguage string `json:"Content-Language" example:"en-US"`
+}
+
+type HwSubPayload struct {
+	Protocol    string       `json:"Protocol" example:"Redfish"`
+	Context     string       `json:"Context"`
+	Destination string       `json:"Destination" example:"http://localhost:8080/api/cloudNotifications/v1/create/event"`
+	EventTypes  []string     `json:"EventTypes" example:["StatusChange", "ResourceUpdated", "ResourceAdded", "ResourceRemoved", "Alert"]`
+	HttpHeaders []HttpHeader `json:"HttpHeaders"`
+}
+
+// Start hw event plugin to process events,metrics and status, expects rest api available to create publisher and subscriptions
+func Start(wg *sync.WaitGroup, scConfig *common.SCConfiguration, fn func(e cneevent.Event) error) error { //nolint:deadcode,unused
+
+	status, _ := createHwEventSubscription(hwEventTypes)
+	if status != http.StatusCreated {
+		err := fmt.Errorf("hw event subscription creation api at %s, returned status %d", hwSubUrl, status)
+		return err
+	}
+
+	//channel for the transport handler subscribed to get and set events
+	//eventInCh := make(chan *channel.DataChan, 10)
+	pubSubInstance := v1pubsub.GetAPIInstance(".")
+	endpointURL := &types.URI{URL: url.URL{Scheme: "http", Host: localAPIAddr, Path: localAPIPath}}
+	// create publisher
+	pub, err := pubSubInstance.CreatePublisher(v1pubsub.NewPubSub(endpointURL, resourceAddress))
+
+	if err != nil {
+		log.Errorf("failed to create a publisher %v", err)
+		return err
+	}
+	log.Printf("Created publisher %v", pub)
+
+	// once the publisher response is received, create a transport sender object to send events.
+	v1amqp.CreateSender(scConfig.EventInCh, pub.GetResource())
+
+	// serve ack/event event response
+	go server()
```

Comment:
Do we need a webserver here in the plugin? you can use default rest API  event ack route